src/components/vm/snapshots/vmSnapshotsCreateModal.jsx

```diff
@@ -139,29 +194,38 @@ export class CreateSnapshotModal extends React.Component {
         else if (!name)
             validationError.name = _("Name can not be empty");
 
-        if (isExternal && vm.state === "running" && !memoryPath)
-            validationError.memory = _("Memory file can not be empty");
+        if (isExternal && vm.state === "running" && !memoryLocation)
+            validationError.memory = _("Memory save location can not be empty");
 
         return validationError;
     }
 
     onCreate() {
         const Dialogs = this.context;
         const { vm, isExternal } = this.props;
-        const { name, description, memoryPath } = this.state;
+        const { name, description, memoryLocation } = this.state;
         const validationError = this.onValidate();
 
         if (!Object.keys(validationError).length) {
             this.setState({ inProgress: true });
-            snapshotCreate({
-                vm,
-                name,
-                description,
-                isExternal,
-                memoryPath: isExternal && vm.state === "running" && memoryPath,
-            })
+            let mpath = null;
+            if (isExternal && vm.state === "running" && memoryLocation) {
+                mpath = memoryLocation;
+                if (mpath[mpath.length - 1] != "/")
+                    mpath = mpath + "/";
+                mpath = mpath + vm.name + "." + name + ".save";
+            }
+            cockpit.spawn(["mkdir", "-p", memoryLocation], { superuser: "try", err: "message" })
+                    .then(() =>
+                        snapshotCreate({
+                            vm,
+                            name,
+                            description,
+                            isExternal,
+                            memoryPath: mpath,
+                        }))
                     .then(() => {
-                        // VM Snapshots do not trigger any events so we have to refresh them manually
+                    // VM Snapshots do not trigger any events so we have to refresh them manually
```

Comment:
Oopsie, please revert (where's our linters for such stuff?)

Reply:
Done!